File: project/app/src/main/java/com/marcosvbras/empresas/models/api/refrofit/APIEndpoints.java
package com.marcosvbras.empresas.models.api.refrofit;

import com.marcosvbras.empresas.models.api.responses.EnterpriseDetailResponse;
import com.marcosvbras.empresas.models.api.responses.EnterpriseResponse;
import com.marcosvbras.empresas.models.api.responses.LoginBody;
import com.marcosvbras.empresas.views.utils.Constants;

import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface APIEndpoints {

    @Headers(Constants.JSON_CONTENT_TYPE_HEADER)
    @POST(Constants.SIGN_IN_ENDPOINT)
    Observable<Void> login(@Body LoginBody login);

    @GET(Constants.ENTERPRISES_LIST_ENDPOINT)
    Observable<EnterpriseResponse> getAllEnterprises();

    @GET(Constants.FILTER_ENTERPRISE_ENDPOINT)
    Observable<EnterpriseResponse> getEnterprises(@Query("name") String name);

    @GET(Constants.GET_ENTERPRISE_ENDPOINT)
    Observable<EnterpriseDetailResponse> getEnterprise(@Path("id") int id);

}

File: project/app/src/main/java/com/marcosvbras/empresas/business/api/refrofit/APIEndpoints.kt
package com.marcosvbras.empresas.business.api.refrofit

import com.marcosvbras.empresas.business.api.responses.EnterpriseDetailResponse
import com.marcosvbras.empresas.business.api.responses.EnterpriseResponse
import com.marcosvbras.empresas.business.api.responses.LoginBody
import com.marcosvbras.empresas.utils.*
import io.reactivex.Observable
import retrofit2.http.*

interface APIEndpoints {

    @Headers(JSON_CONTENT_TYPE_HEADER)
    @POST(SIGN_IN_ENDPOINT)
    fun login(loginBody: LoginBody) : Observable<Void>

    @GET(ENTERPRISES_LIST_ENDPOINT)
    fun getAllEnterprises() : Observable<EnterpriseResponse>

    @GET(FILTER_ENTERPRISE_ENDPOINT)
    fun getEnterprises(@Query("name") name: String) : Observable<EnterpriseResponse>

    @GET(GET_ENTERPRISE_ENDPOINT)
    fun getEnterprise(@Path("id") id: Int) : Observable<EnterpriseDetailResponse>
}